writer.go

```diff
@@ -0,0 +1,383 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dictzip
+
+import (
+	"bytes"
+	"compress/flate"
+	"encoding/binary"
+	"fmt"
+	"hash"
+	"hash/crc32"
+	"io"
+	"math"
+	"os"
+	"time"
+)
+
+const (
+	DefaultChunkSize = math.MaxUint16
+)
+
+const (
+	NoCompression      = flate.NoCompression
+	BestSpeed          = flate.BestSpeed
+	BestCompression    = flate.BestCompression
+	DefaultCompression = flate.DefaultCompression
+	HuffmanOnly        = flate.HuffmanOnly
+)
+
+// Writer implements [io.WriteCloser] for writing dictzip files. Writer writes
+// chunks to a temporary file during write and copies the resulting data to the
+// final file when [Writer.Close] is called.
+//
+// For this reason, [Writer.Close] must be called in order to write the file
+// correctly.
```

Comment:
Add a note to `dictzip.go` that these implementations are not safe for parallel execution.